omegaconf/grammar/OmegaConfGrammarLexer.g4

```diff
@@ -33,6 +33,8 @@ mode VALUE_MODE;
 INTER_OPEN: '${' WS? -> pushMode(INTERPOLATION_MODE);
 BRACE_OPEN: '{' WS? -> pushMode(VALUE_MODE);  // must keep track of braces to detect end of interpolation
 BRACE_CLOSE: WS? '}' -> popMode;
+QUOTE_OPEN: '\'' -> pushMode(QUOTED_SINGLE_MODE);
+QUOTE_DOUBLE: '"' -> type(QUOTE_OPEN), pushMode(QUOTED_DOUBLE_MODE);
```

Comment:
Naming is a bit inconsistent.
Can you make the two quote types symmetric?

Reply:
So I can see a few different options here -- let me know what you prefer (or if you have other ideas):
1. The current one I went with
2. Renaming `QUOTE_OPEN` into `QUOTE_SINGLE`: the problem is that it would look weird in the parser rule `quotedValue` (`QUOTE_SINGLE ... QUOTE_CLOSE`)
3. Using two lexer tokens `QUOTE_OPEN_SINGLE` and `QUOTE_OPEN_DOUBLE` instead of a single `QUOTE_OPEN` -- this would turn the parser rule `quotedValue` into `(QUOTE_OPEN_SINGLE | QUOTE_OPEN_DOUBLE) ... QUOTE_CLOSE`
4. Creating a dummy mode just to define a `QUOTE_OPEN` token so that I can write `QUOTE_SINGLE: '\'' -> type(QUOTE_OPEN), pushMode(QUOTED_SINGLE_MODE);` (I can't do that without a `QUOTE_OPEN` token defined elsewhere, ANTLR doesn't allow it otherwise)

Reply:
not sure I fully understand 4, but 3 sounds good.
We can also potentially rename QUOTE_CLOSE to MATCHING_QUOTE_CLOSE.

Reply:
So:
* I did option 3 in 50cb0c6 just so that you can see what it means
* I reverted it right after and did 4 in ef888e3 (let me know if in the end you prefer 3)
* I renamed QUOTE_CLOSE in f85bdee

Reply:
Cool.
I like 3, feels a bit more straight forward than 4 even if it's a bit more verbose.

Reply:
Alright, sticking with 3 then (4 will get squashed away during merge) => resolving